### PR TITLE
feat: Add service user guide

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: Harrison <harrisonliu_5@163.com>

### CODE REVIEW

**Comment by Sherlock113** [comment]
Is this pull request finished? I saw the title still has "WIP" in it. If it is finished, I will review it.

**Comment by harrisonliu5** [comment]
please review the stateful first, because they have some part as the same @Sherlock113

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 1]
```diff
@@ -1,10 +1,167 @@
...
+
+  The service created by External Service is a service internet access method. It is different from Stateless and stateful.
+
+- **Specify Workloads**
+
+  Create a service with existing Pods.
+
+- **Edit by YAML**
+
+  Create directly with Yaml, you can upload and download Yaml configuration files
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:64]: ->
Create a service directly with YAML. You can upload and download YAML configuration files to and from the console.

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 2]
```diff
@@ -1,10 +1,167 @@
...
+
+  The value of `annotations:kubesphere.io/serviceType` keywords is one of them **statelessservice** , **statefulservice** , **externalservice** and **None**.
+
+{{</ notice>}}
+
+## Create a **Stateless Service**
+
+Into **Application Workload**, click **Services**
+
+### Step 1: click the **Create**, choose **Stateless Service**
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:76]: ### Step 1: click the **Create**, choose **Stateless Service**
->
### Step 1: click **Create**, and choose **Stateless Service**

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 3]
```diff
@@ -1,10 +1,167 @@
...
+
+![stateless_form_1](/images/docs/project-user-guide/workloads/stateless_form_1.png)
+
+- **Name**: The name of the Service and Deployment, which is also the unique identifier.
+- **Alias**: The alias name of the service, making resources easier to identify.
+- **Version**: It can only contain lowercase letters and numbers. The maximum length of characters is set to 16.
+
+{{< notice tip >}}
+
+**Name** have named two configs, `Deployment` and `Service`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:92]: Sorry, I don't quite understand this sentence...I guess you mean the value of `kind` in YAML? Two configurations in the yaml file: one `Deployment` and the other one is `Service`, both of which contain the value of **Name**. If so, please change it to:

The value of **Name** is used in both configurations, one for `Deployment` and the other for `Service`.

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 4]
```diff
@@ -1,10 +1,167 @@
...
+
+Refer to [Deployment Advanced Settings](../deployments) for details.
+
+## Check Service Detail
+
+### Service Operations
+
+![stateless_finish](/images/docs/project-user-guide/workloads/services_lists_finish.png)
+
+Now the Stateless Service has created, you can click it to view more information about it.also you can click **More** to operate this service item include edit config and delete.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:151]: ->
After the stateless service is created, you can click the three dots on the right to further edit it, such as its metadata (excluding **Name**), YAML, port, and Internet access.

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 5]
```diff
@@ -1,10 +1,167 @@
...
+- **Edit YAML**: Edit the YAML config.
+- **Edit Service**: Edit the service config.
+- **Edit Internet Access**: Edit the service internet access.
+- **Delete**: When deleting a service, the associated resources will be displayed. If you check them, they will be deleted together with the service
+
+### Service Detail
+
+![stateless_finish](/images/docs/project-user-guide/workloads/services_detail.png)
+
+- The **More** Operations is the same which in the services list.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:163]: ->
Click **More** to expand the drop-down menu which is the same as the one in the service list.

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 6]
```diff
@@ -1,10 +1,167 @@
...
+- The pod list provides the pod's detail information(conditions, phase, node, pod ip, monitoring).
+- You can view the container info by click the pod item.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:165]: by click
->
by clicking

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 7]
```diff
@@ -1,10 +1,167 @@
...
+- Click the container log icon to view the output logs of the container.
+- You can view the pod detail page by click the pod name.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/services.md:167]: by click
->
by clicking

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments and make changes. Thanks.

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 8]
```diff
@@ -1,10 +1,167 @@
...
+## Service Type
+
+- **VirtualIP**：It is based on the unique IP generated by the cluster. The service inside the cluster can be accessed through this IP, and the service outside the cluster can be accessed through NodePort and LoadBalancer. This type is suitable for most services.
+- **Headless**：The cluster does not generate an IP for the service, and the service is directly accessed through the back-end Pod IP of the service within the cluster. This type is suitable for back-end heterogeneous services, such as services that need to distinguish between master and slave.
+
+{{< notice tip>}}
+
+In Kubesphere, the **Virtual IP** is a default type of a service that is created by **stateless** or **stateful**.<br />
+
+If you want to create a Headless service, you should use the mode of **YAML**.
```
[the remark this answers is not in the document]
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 9]
```diff
@@ -1,10 +1,167 @@
...
+## Prerequisites
+
+- You need to create a workspace, project and `project-regular` account. Please refer to the [Getting Started with Multi-tenant Management](../../../quick-start/create-workspace-and-project) if not yet.
+- You need to sign in with `project-admin` account and invite `project-regular` to enter the corresponding project if not yet. Please refer to [Invite Member](../../../quick-start/create-workspace-and-project#task-3-create-a-project).
+
+## How to create a Service
+
+![create_service_type](/images/docs/project-user-guide/workloads/create_service_type.png)
+
+Create a service that has these methods in the Kubesphere: **Stateless Service**, **Stateful Service**, and **External Service**. what's more, you can custom creation through **Specify Workloads** and **Edit by YAML**.
```
[the remark this answers is not in the document]
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 10]
```diff
@@ -1,10 +1,167 @@
...
+
+There are some ways to create a service
+
+- **Stateless Service**
+
+  Create a Stateless Service also create a **Deployment** workload.
+
+  For more information **Stateless**, please visit [Deployment](../../application-workloads/deployments)
```
[the remark this answers is not in the document]
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 11]
```diff
@@ -1,10 +1,167 @@
...
+
+- **Stateful Service**
+
+  Create a Stateless Service also create a **Statefulset** workload.
+
+  For more information **Statefulset**, please visit [Statefulset](../../application-workloads/statefulsets)
```
[the remark this answers is not in the document]
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md [hunk 12]
```diff
@@ -1,10 +1,167 @@
...
+
+- **External Service**
+
+  The service created by External Service is a service internet access method. It is different from Stateless and stateful.
```
[the remark this answers is not in the document]
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 1 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 2 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 3 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 4 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 5 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 6 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

File: content/en/docs/project-user-guide/application-workloads/services.md (hunk 7 above)
Comment (harrisonliu5) [reply content/en/docs/project-user-guide/application-workloads/services.md]: done

**Comment by Sherlock113** [comment]
/lgtm
@FeynmanZhou Please merge this pr and I will update the entire wording and format. Thanks.

**Comment by FeynmanZhou** [comment]
@Sherlock113 I will merge this PR, please update the weight should be adjusted accordingly.

image

**Pull request merged**